recipes/error-handling.md

```diff
@@ -0,0 +1,34 @@
+---
+description: How error messages can lead to security issues ?
+complexity: 1
+---
+
+# Handling error messages
+
+## Use-case
+
+Understanding error messages is an important part of web application development, However revealing too many details can allow attackers to uncover vulnerabilities that can be used to exploit system flaws. We should properly handle our error messages and ensure we don't expose sensitive data to our users.
+
+
+## Recipe
+
+When a user tries to access a resource that does not exist, the first step is to provide a proper status code, `404 not found` in this case. Additionally, you may need to provide more information in the response body, but be careful about the details you add to your error message ! 
+
+Let's take the following example : 
+
+`GET /v1/clients/12345/attributions/21100512345`
+=> 404 "The Attribution 211005123456 was not found"
```

Comment:
je veux bien qu'on "anonymise" l'exemple : celui-là est tiré de l'API de la Compagnie des Alpes (qui n'est pas publique), et en plus "attribution" n'est pas très parlant quand on n'a pas le contexte métier

tu peux remplacer `attributions` par `orders`, ce sera plus générique


autre suggestion : simplifier carrément en ne mettant que `orders` (la partie "client linked to" est un peu tirée par les cheveux)

ça donnerait :
```
`GET /orders/21100512345` => 404

`GET /orders/21100534871` => 403 => donc cet id existe
```